azure/src/wizard/StorageAccountCreateStep.ts

```diff
@@ -47,6 +50,59 @@ export class StorageAccountCreateStep<T extends types.IStorageAccountWizardConte
         ext.outputChannel.appendLog(createdStorageAccount);
     }
 
+    protected async generateRelatedName(wizardContext: T, name: string, namingRules: IAzureNamingRules | IAzureNamingRules[]): Promise<string | undefined> {
+        if (!Array.isArray(namingRules)) {
+            namingRules = [namingRules];
+        }
+
+        let preferredName: string = namingRules.some((n: IAzureNamingRules) => !!n.lowercaseOnly) ? name.toLowerCase() : name;
+
+        for (let invalidCharsRegExp of namingRules.map((n: IAzureNamingRules) => n.invalidCharsRegExp)) {
+            // Ensure the regExp uses the 'g' flag to replace _all_ invalid characters
+            invalidCharsRegExp = new RegExp(invalidCharsRegExp, 'g');
+            preferredName = preferredName.replace(invalidCharsRegExp, '');
+        }
+
+        const minLength: number = Math.max(...namingRules.map((n: IAzureNamingRules) => n.minLength));
+        const maxLength: number = Math.min(...namingRules.map((n: IAzureNamingRules) => n.maxLength));
+
+        const maxTries: number = 5;
+        let count: number = 0;
+        let newName: string;
+        while (count < maxTries) {
+            newName = this.generateSuffixedName(preferredName, minLength, maxLength);
+            if (await this.isRelatedNameAvailable(wizardContext, newName)) {
+                return newName;
+            }
+            count += 1;
+        }
+
+        const createdStorageAccount: string = localize('FailStorageAccount', 'Something went wrong in the creation of the storage account');
```

Comment:
Sorry, but could you remove this line? I know I mentioned that it'd be good to catch the error, but with this implementation, they'll see this error message in output, but when this line runs:
```
wizardContext.storageAccount = await storageClient.storageAccounts.beginCreateAndWait(
            // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
            wizardContext.resourceGroup!.name!,
            newName,
            {
                sku: { name: newSkuName },
                kind: this._defaults.kind,
                location: newLocation,
                enableHttpsTrafficOnly: true
            }
        );
```
it'll throw another error, probably something about how the name is invalid or something (since it'll be undefined).

I'm not super worried about this actually happening with a randomHexString as a suffix, so we can probably just go ahead and remove these lines.